templates/commands/render/render.go

```diff
@@ -495,10 +510,40 @@ func checkInputsMissing(spec *spec.Spec, inputs map[string]string) []string {
 func executeSteps(ctx context.Context, steps []*spec.Step, sp *stepParams) error {
 	logger := logging.FromContext(ctx).With("logger", "executeSteps")
 
+	if sp.flags.DebugStepDiffs {
+		argsList := [][]string{
+			// Make debug dir a git repository with a detached work tree in scratch
+			// dir, meaning it will track the file changes in scratch dir without
+			// affecting the scratch dir.
+			strings.Fields(fmt.Sprintf(
+				"--git-dir=%s --work-tree=%s init", sp.debugDir, sp.scratchDir)),
```

Comment:
git -C <pathA> init <pathB> will make <pathA> a git repo, not "pathB as git repo but .git stored in pathA".